File: FrontEnd/src/services/logisticsService.ts
import { DeliveryMode, DeliveryQuote } from '../types/domain';

const modeBase: Record<DeliveryMode, number> = {
  pickup: 0,
  seller_delivery: 54,
  partner_delivery: 68,
};

export const logisticsService = {
  calculate(distanceKm: number, volume: number, mode: DeliveryMode): DeliveryQuote {
    if (mode === 'pickup') {
      return {
        distanceKm,
        volume,
        mode,
        price: 0,
        etaDays: 1,
      };
    }

    const raw = distanceKm * modeBase[mode] + volume * 120;
    const price = Math.round(raw);
    const etaDays = Math.max(1, Math.ceil(distanceKm / 250));

    return { distanceKm, volume, mode, price, etaDays };
  },
};

File: Backend/src/Zerno.Infrastructure/Services/LogisticsService.cs
using Zerno.Application.Abstractions;
using Zerno.Application.Contracts.Logistics;

namespace Zerno.Infrastructure.Services;

public sealed class LogisticsService : ILogisticsService
{
    public Task<DeliveryQuoteDto> CalculateAsync(DeliveryQuoteRequestDto request, CancellationToken cancellationToken)
    {
        var modeMultiplier = request.Mode.Trim().ToLowerInvariant() switch
        {
            "pickup" => 0.75m,
            "seller_delivery" => 1.0m,
            "partner_delivery" => 1.18m,
            _ => 1.0m
        };

        var price = Math.Round((request.DistanceKm * 22m + request.Volume * 8m) * modeMultiplier, 2);
        var etaDays = request.DistanceKm <= 100 ? 1 : request.DistanceKm <= 400 ? 3 : 6;
        return Task.FromResult(new DeliveryQuoteDto(request.DistanceKm, request.Volume, request.Mode, price, etaDays));
    }
}
